src/compiler/checker.ts

```diff
@@ -9779,14 +9779,22 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     if (p.flags & SymbolFlags.Accessor && useAccessors) {
                         const result: AccessorDeclaration[] = [];
                         if (p.flags & SymbolFlags.SetAccessor) {
+                            const setter = p.declarations?.find((d): d is SignatureDeclaration => {
+                                if (d.kind === SyntaxKind.SetAccessor) {
+                                    return true;
+                                }
+                                const id = getNameOfDeclaration(d);
+                                return !!id && isIdentifier(id) && idText(id) === "set";
+                            });
+                            const paramSymbol = setter && getSignatureFromDeclaration(setter)?.parameters[0];
                             result.push(setTextRange(
                                 factory.createSetAccessorDeclaration(
                                     factory.createModifiersFromModifierFlags(flag),
                                     name,
                                     [factory.createParameterDeclaration(

```

Comment:
This whole call could just be replaced with `symbolToParameterDeclaration(paramSymbol, context)` but types don't line up here. `p.declarations` is nullish - how safe, in general, it is to assume that nodes here have `declarations`, or `valueDeclaration` and similar things? I could just use a non-nullish assertion but I'm not sure if that's a good idea here - especially since this code was working without relying on those things before.

Reply:
By the time it gets here, is it possible for it to be `undefined` at all? e.g. could you assert it?

Reply:
Hm, the answer seems like "no", and it's also incorrect for setters like:

```
    /** @param {string=} value */
    set p1(value: string | undefined);
```

As the code "correctly" handles the `=` and then makes it optional, when that's invalid. Though that could be fixed.

Maybe just a new helper that uses a fixed `symbolToParameterDeclaration` if the symbol exists, otherwise falls back to a generated name?

Reply:
> Hm, the answer seems like "no"

No - it isn't possible for it to be `undefined` here. Or no - I can't assert it here? 😅 

> As the code "correctly" handles the = and then makes it optional, when that's invalid. Though that could be fixed.

Could you rephrase?

Reply:
No as in a test fails if it's asserted, as it can be undefined.

I think it's probably clearer if you make the change and see the test change; there's a d.ts file which turns out invalid if you just use the other function as is.